pkgpanda/util.py

```diff
@@ -29,6 +31,70 @@
 }
 
 
+def is_absolute_path(path):
+    if is_windows:
+        # We assume one char drive letter. Sometimes its two but not often
+        # pattern is <driveletter>:/string....
+        if path[1] == ':':
+            return True
+    else:
+        if path[0] == '/':
+            return True
+    return False
+
+
+if is_windows:
```

Comment:
sorry to distress you. I will change that. 
I have talked about these functions before in detail and often they break, especially the rm and copy. I think I have had problems with makedirs too.
This solution of calling into the underlying utilities (much like you did) works and so we stuck with that. 
I would prefer to stick with the current solution because after the significant amount of time we have already spent on this, this solution really does work.

Reply:
I also spent time looking on the web about some of these solutions and many times people tried and failed, much like we did.

Reply:
SGTM. Let's just add a comment here explaining why we're implementing our own helpers instead of using the ones from stdlib. And these new helpers should be straightforward to test, so let's add a few unit tests in `pkgpanda/test_util.py` that cover them. We'd want to set up files in a tmpdir and try to copy, move, etc. and verify that the result is what we expect. You can grep the repo for `tmpdir` to see examples of tests doing similar things, for reference.

Reply:
I relocated code from elsewhere into the helper code and you want me to create unittests around that.
I duplicated the style of functionality as you with regards to calling an external command and you want me to use internal commands.
Why are you setting a higher standard for me over what you did yourselves? I don't get that. I have added comments to the windows versions of the code as you suggested.

Reply:
Most of the helpers in this file are covered by unit tests, and this project heavily values testing, so it's not a new standard. And those tests would not be complicated to write. Create a tmpdir, use the function under test to manipulate it, verify the output is what you expect.

Reply:
But at your request I added these functions thereby I was just duplicating what you did in code. Why do I want to test rm -rf on Linux or del /s on windows. We know they work. At the same time we know os.makedir works otherwise it would not be a system library. All this seems like very much overkill and takes my time away from porting pkgpanda to windows.
If you want you can take over the porting while I write unittests?

Reply:
When we add new units, we try to add new unit tests. You're adding new functions that have new functionality: cross-platform behavior. These functions are expected to behave the same on each platform, so that's what we'd want to test. We'd want to add platform-independent tests that run on both Linux and Windows, and verify that the same inputs yield the expected outputs (with some exceptions for e.g. filesystem path formatting) on either platform.

Reply:
but if I had left this code where you originally just spawned an external command there would be no tests other than the code working, which it does.
Testing the rm -rf with all sorts of test cases to catch failure and success is larger than you think. Maybe easy for you who knows python, not so much for me who does not know python.
Again, doing this is going to stop me working on porting pkgpanda and making the work in progress a final product.